src/test/java/teammates/test/cases/webapi/SearchStudentsActionTest.java

```diff
@@ -0,0 +1,145 @@
+package teammates.test.cases.webapi;
+
+import org.junit.Test;
+
+import teammates.common.datatransfer.DataBundle;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.util.Const;
+import teammates.ui.webapi.action.JsonResult;
+import teammates.ui.webapi.action.SearchStudentsAction;
+import teammates.ui.webapi.output.SearchStudentsData;
+
+/**
+ * SUT:{@link SearchStudentsAction}.
+ */
+public class SearchStudentsActionTest extends BaseActionTest<SearchStudentsAction> {
+
+    @Override
+    protected void prepareTestData() {
+        DataBundle dataBundle = getTypicalDataBundle();
+        removeAndRestoreDataBundle(dataBundle);
+        putDocuments(dataBundle);
+    }
+
+    @Override
+    protected String getActionUri() {
+        return Const.ResourceURIs.STUDENTS_SEARCH;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return GET;
+    }
+
+    @Override
+    protected void testExecute() {
+        // TODO: Remove
+    }
+
+    @Test
+    public void execute_notEnoughParameters_parameterFailure() {
+        loginAsAdmin();
+        verifyHttpParameterFailure();
+    }
+
+    @Test
+    public void execute_adminSearchName_success() {
+        StudentAttributes acc = typicalBundle.students.get("student1InCourse1");
+        loginAsAdmin();
+        String[] accNameParams = new String[] {
+                Const.ParamsNames.SEARCH_KEY, acc.getName(),
+        };
+        SearchStudentsAction a = getAction(accNameParams);
+        JsonResult result = getJsonResult(a);
+        SearchStudentsData response = (SearchStudentsData) result.getOutput();
+        assertEquals(11, response.getStudents().size());
```

Comment:
@hcwong I think hardcoding the value is okay if the expected value is 0 (in the case of no match) but for the rest, it's a bit fragile - everything will probably break if the typical bundle data changes. What do you think about something like this: https://github.com/dvrylc/teammates/pull/2/files#diff-4b62c9dcdd4b81207d24a3a3ba7ac97eR55?

Reply:
Sure lets do this